cfsdk/src/main/java/io/harness/cfsdk/cloud/oksse/SSEListener.java

```diff
@@ -62,29 +62,32 @@ public void onMessage(ServerSentEvent serverSentEvent, String id, String event,
                     )
             );
 
-            if (
-                    ("delete".equals(eventType) || "patch".equals(eventType))  &&
-                            "target-segment".equals(domain)
-            ) {
-
-                final StatusEvent statusEvent =
-                        new StatusEvent(StatusEvent.EVENT_TYPE.EVALUATION_RELOAD, evaluation);
-
-                eventsListener.onEventReceived(statusEvent);
-
-            } else if ("create".equals(eventType) || "patch".equals(eventType)) {
-
-                final StatusEvent statusEvent =
-                        new StatusEvent(StatusEvent.EVENT_TYPE.EVALUATION_CHANGE, evaluation);
-
-                eventsListener.onEventReceived(statusEvent);
-
-            } else if ("delete".equals(eventType)) {
+            StatusEvent statusEvent = null;
+            if ("target-segment".equals(domain)) {
+                // On creation, change or removal of a target group we want to reload evaluations
+                if ("delete".equals(eventType) || "patch".equals(eventType) || "create".equals(eventType)) {
+                    statusEvent = new StatusEvent(StatusEvent.EVENT_TYPE.EVALUATION_RELOAD, evaluation);
+                }
+            }
 
-                final StatusEvent statusEvent =
-                        new StatusEvent(StatusEvent.EVENT_TYPE.EVALUATION_REMOVE, evaluation);
+            if ("flag".equals(domain)) {
+                // On creation or change of a flag we want to send a change event
+                if ("create".equals(eventType) || "patch".equals(eventType)) {
+                    statusEvent = new StatusEvent(StatusEvent.EVENT_TYPE.EVALUATION_CHANGE, evaluation);
+                }
+                // On deletion of a flag we want to send a remove event
+                if ("delete".equals(eventType)) {
+                    statusEvent = new StatusEvent(StatusEvent.EVENT_TYPE.EVALUATION_REMOVE, evaluation);
```

Comment:
I think we looked at this when going through the code originally.  On delete does the evaluation object just contain the identifier and kind, with the value null?

Reply:
Yeah exactly that